### ISSUE TITLE
False Negative: NN_NAKED_NOTIFY is incorrectly suppressed by getfield and invokevirtual instructions

### ISSUE THREAD

**Issue by leemeii** [spotbugs/spotbugs#3884]
**Describe the bug**
The **NN_NAKED_NOTIFY** rule fails to report a naked notify when the synchronized block contains certain non-state-changing instructions (e.g., field reads or method calls).

```java
public class Bug {
    private int count = 0;

//Case 1 
    public synchronized void testWithFieldRead() {
        int temp = this.count; 
        this.notifyAll();  // <- should report 
    }
//Case 2
    public synchronized void testWithPrintln() {
        System.out.println("Log"); 
        this.notifyAll();  // <- should report 
    }
}
```

**Expected behavior**
Reading a variable or printing a log message does not modify the object's state.
Therefore, the notification remains "naked" because any awakened thread will find the condition unchanged and return to the waiting state immediately.

**Actual behavior**
Both of them are ignored.

**Environment**
- Java Compiler: JDK 21
- Java version:  JDK 21
- SpotBugs Version:  4.9.8
- How do you use SpotBugs: CLI

**Additional context**
There is a significant logical inconsistency in Case 1. SpotBugs successfully reports DLS_DEAD_LOCAL_STORE for the variable temp in testWithFieldRead. 
This proves the analysis engine knows the getfield instruction has no meaningful impact on the program's logic. However, this same "dead" instruction is allowed to suppress the NN_NAKED_NOTIFY concurrency warning.
